serverless/experimental/wasm/main.go

```diff
@@ -209,29 +236,39 @@ func monitor(ctx context.Context, f client.Fetcher) {
 		if cp.Size <= cpCur.Size {
 			continue
 		}
-		monMsg("----------------------------------")
 		monMsg(fmt.Sprintf("<invert>Saw new CP with size %d</invert>", cp.Size))
 
-		if cpCur.Size > 0 {
-			pb, err := client.NewProofBuilder(ctx, *cp, rfc6962.DefaultHasher.HashChildren, f)
-			if err != nil {
-				monMsg(err)
-				continue
-			}
-			proof, err := pb.ConsistencyProof(ctx, cpCur.Size, cp.Size)
+		rCP := rf.NewEmptyRange(cpCur.Size)
+		for i := cpCur.Size; i < cp.Size; i++ {
+			l, err := client.GetLeaf(ctx, f, i)
 			if err != nil {
-				monMsg(err)
-				continue
+				monMsg(fmt.Sprintf("Failed to fetch leaf at %d: %v", i, err))
+				break monitorLoop
```

Comment:
`return` would be more idiomatic and remove the need for the label?

Reply:
That would kill the monitor though?
The break takes us back up to sleep and try again